src/deepsparse/v2/pipeline.py

```diff
@@ -13,88 +13,113 @@
 # limitations under the License.
 
 
-from typing import List
-
-from pydantic import BaseModel, Field, PrivateAttr
+from typing import Any, Dict, List, Optional, Union
 
 from deepsparse.v2.operators import Operator
 from deepsparse.v2.routers import Router
 from deepsparse.v2.schedulers import OperatorScheduler, SchedulerGroup
+from deepsparse.v2.utils import Context
 
 
 __all__ = ["Pipeline"]
 
 
-class Pipeline(BaseModel):
-    """
-    Pipeline accepts a series of operators, schedulers, and a router which define
-    an end to end ML transformation.
-
-    Calling a pipeline runs these transformations
+class Pipeline(Operator):
     """
+    Pipeline accepts a series of operators, schedulers, and a router. Calling a pipeline
```

Comment:
I'd expand docstrings here a bit to provide the users with some fundamental info on operators, schedules, routers, and their relationships. Not a full-blown documentation, but something to give the reader an initial idea of what those are.